### ISSUE TITLE
Unable to upload to hubiC - The object was used after being disposed

### ISSUE THREAD

**Issue by steilerDev** [duplicati/duplicati#1431]
**Steps to reproduce**: 
1. [Download and install duplicati](http://www.duplicati.com/howtos/how-to-install-and-run-duplicati-2-0-preview)
2. Get HubiC account and [obtain OauthToken](https://duplicati-oauth-handler.appspot.com/type=hubic)
3. Start backup using: `mono Duplicati.CommandLine.exe backup "hubic:///Backup" "/media/files/Pictures" --authid="<auth-token>" --backup-name="Photos" --dbpath="/opt/duplicati/config/HUBIC.sqlite" --encryption-module="aes" --keep-time="3M" --passphrase="secret" --dblock-size="100mb"`

**Expected result**: Files are uploaded to the server.

**Actual result**: Program exits unexpected, not uploading any files with the following message: 
`Operation Put with file duplicati-b721e05bcb8ee4236a2d164d6bfc3599c.dblock.zip.aes attempt 1 of 5 failed with message: The object was used after being disposed. => The object was used after being disposed.
Renaming "duplicati-b721e05bcb8ee4236a2d164d6bfc3599c.dblock.zip.aes" to "duplicati-b4350c9964955453382d7e109efb16d6f.dblock.zip.aes"

Edit: I created the `Backup` folder in my hubiC account.
##

**Comment by dannysu** [comment]
I had similar error message before and it was before mono v4 imported a bunch Microsoft code and certain things weren't 100% compatible. I think the fix should be available if you have mono 4.0.3.20, but if not, stick to mono 3 should fix it.

**Comment by kenkendk** [comment]
The error message is hiding the real error, making it hard to say what goes wrong.

In some cases, this is caused by SSL issues, which can be fixed with mozroots:
http://www.mono-project.com/docs/faq/security/

tl;dr; run: ```sudo mozroots --import --sync```

**Comment by steilerDev** [comment]
@dannysu @kenkendk sorry for taking so long, but I did not have access to the server running duplicati until now.

Unfortunately `sudo mozroots --import --sync` did not work. I will try and install mono 3 (I updated to 4.4 earlier).

**Comment by kenkendk** [comment]
I was just looking at mozroots again, and maybe you need to add the `--machine` flag also:
`sudo mozroots --import --sync --machine`, otherwise the certificates might end up in the root account only.

**Comment by steilerDev** [comment]
Nope did not do the trick either, especially since I was running the backup as root anyway.

Additionally the TLS check from your link does not indicate a problem with hubic.com.

I think it might be something different, any other suggestions @kenkendk?

**Comment by kenkendk** [comment]
Was this resolved?

Otherwise, could you try with adding `--verbose` and reporting the full error message.
I saw a similar error message that was fixed by setting `--dblock-size=10mb`.

**Comment by steilerDev** [comment]
I moved away from hubiC since it made a lot of problems and I never got it working with duplicati. So I don't know. Sorry!

**Comment by wassname** [comment]
Your advice fixed the issue for me thanks (block size 10mb, turn on ssl, `sudo mozroots --import --sync --machine`)
